src/components/fileStorage/repo/files.repo.js

```diff
@@ -0,0 +1,41 @@
+const { FileModel } = require('./db');
+
+/**
+ * @param {*} fileIds
+ * @param {*} userId
+ * @return {MongooseBatchResult}
+ */
+const removeFilePermissionsByUserId = async (userId) => {
+	const searchQuery = { permissions: { $elemMatch: { refId: userId } } };
+
+	const filePermissions = await FileModel.aggregate([
+		{
+			$match: {
+				permissions: {
+					$elemMatch: { refId: userId },
```

Comment:
Should only match permissions where the user is not the owner of the file. To optimize this where the open task in the old code and the test for it, already exist.

Reply:
once we are deleting the diles where the user is the owner, these will already be gone at this point.